common/get_prices.go

```diff
@@ -0,0 +1,130 @@
+package common
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+// MainnetEndpoint is a mainnet Ethereum endpoint.
+// from https://chainlist.org/chain/1
+const MainnetEndpoint = "https://eth-rpc.gateway.pokt.network"
+
+var (
+	functionSig = ethcrypto.Keccak256([]byte("latestRoundData()"))[:4]
+
+	errUnsupportedNetwork = errors.New("unsupported network")
+)
+
+// see latestRoundData at https://docs.chain.link/data-feeds/price-feeds/api-reference/
+func latestRoundDataReturnArgs() abi.Arguments {
+	uint256Ty, err := abi.NewType("uint256", "", nil)
+	if err != nil {
+		panic(fmt.Errorf("failed to create uint256 type: %w", err))
+	}
+
+	int256Ty, err := abi.NewType("int256", "", nil)
+	if err != nil {
+		panic(fmt.Errorf("failed to create int256 type: %w", err))
+	}
+
+	uint80Ty, err := abi.NewType("uint80", "", nil)
+	if err != nil {
+		panic(fmt.Errorf("failed to create uint80 type: %w", err))
+	}
+
+	return abi.Arguments{
+		{
+			Type: uint80Ty,
+		},
+		{
+			Type: int256Ty,
+		},
+		{
+			Type: uint256Ty,
+		},
+		{
+			Type: uint256Ty,
+		},
+		{
+			Type: uint80Ty,
+		},
+	}
+}
+
+// GetETHUSDPrice returns the current ETH/USD price from the Chainlink oracle.
+// It errors if the chain ID is not the Ethereum mainnet.
+func GetETHUSDPrice(ctx context.Context, ec *ethclient.Client) (*big.Int, error) {
```

Comment:
The description says we are returning the ETH/USD price, but the value we are returning is a `*big.Int`. Can we add information on how the ratio is represented?

I added `t.Logf("ETH/USD price: %s", price)` to `TestGetETHUSDPrice_mainnet` and that gives `116704640000`. Do we need to be returning the number of decimal places as a 2nd parameter? Or is it a constant `8`?

Reply:
it's just the price of ETH in USD, and yes I believe it's a constant 8 decimals